dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskGroupServiceImpl.java

```diff
@@ -439,4 +430,38 @@ public void deleteTaskGroupByProjectCode(long projectCode) {
         taskGroupQueueService.deleteByTaskGroupIds(taskGroupIds);
         taskGroupMapper.deleteBatchIds(taskGroupIds);
     }
+
+    private boolean hasProjectPerm(User loginUser, long projectCode, Map<String, Object> result,
```

Comment:
May I ask why not use `hasProjectAndWritePerm` and `hasProjectAndPerm` in `ProjectServiceImp`?

Reply:
I'm a little confused. `TaskGoupServiceImpl` and `ProjectServiceImpl` are in the same module, why it would cause circular dependency ?

Reply:
It's the circular dependency of Spring Bean creation.